packages/realm-server/migrations/1728561353454_create-user-billing.js

```diff
@@ -0,0 +1,164 @@
+exports.up = (pgm) => {
+  pgm.createTable('users', {
+    id: {
+      type: 'serial',
+      primaryKey: true,
+    },
+    matrix_user_id: { type: 'varchar', notNull: true, unique: true },
+    stripe_customer_id: { type: 'varchar' },
+    created_at: {
+      type: 'timestamp',
+      notNull: true,
+      default: pgm.func('current_timestamp'),
+    },
+  });
+
+  pgm.createTable('plans', {
+    id: {
+      type: 'serial',
+      primaryKey: true,
+    },
+    name: { type: 'varchar', notNull: true },
+    monthly_price: { type: 'numeric(19,4)', notNull: true },
+    credits_included: { type: 'integer', notNull: true },
+    overage_fee_per_credit_in_usd: { type: 'numeric(19,4)' },
+    created_at: {
+      type: 'timestamp',
+      notNull: true,
+      default: pgm.func('current_timestamp'),
+    },
+  });
+
+  pgm.createTable('billing_cycles', {
+    id: {
+      type: 'serial',
+      primaryKey: true,
+    },
+    subscription_id: { type: 'integer' },
+    period_start: { type: 'timestamp', notNull: true },
+    period_end: { type: 'timestamp', notNull: true },
+  });
+
+  pgm.createTable('subscriptions', {
```

Comment:
Free users won't have a subscription. In this case we can get their available credits by querying if their consumed credits in the current month are less than what is allowed (currently 100)

Reply:
We may want/need to consider what happens when a user's renewal payment fails. Do we retry? (I think Stripe does automatically.) Do we want to lock them out immediately? Is there a grace period? How do we reflect this state in the database?

Reply:
Stripe will retry for a couple of days (final retry happens after 7 days). I think we should keep users active in this period. Meaning user's subscription status and billing cycle remains the same during the retry period. We don't have to do anything extra. If subscription expires after the retry period, stripe will inform us and we will end the subscription, or change it to a free plan. If the charge succeeds during the retry period, stripe will send a renewal webhook and we deal with it normally.